utils/atomic_vector.hh

```diff
@@ -62,4 +63,22 @@ public:
             func(_vec[i]);
         }
     }
+
+    // The callback function must not call remove.
+    //
+    // We would take callbacks that take a T&, but we had bugs in the
+    // past with some of those callbacks holding that reference past a
+    // preemption.
+    seastar::future<> for_each_futurized(seastar::noncopyable_function<seastar::future<>(T)> func) {
```

Comment:
I think you can drop _futurized from the name, overload resolution will find the correct overload. If anything the non-future version should have the ugly name.

Reply:
I've tried to drop `_futurized`, but then the compiler complains that it cannot deduce which overload to use in multiple places (i.e. which `seastar::noncopyable_function` instantiation is needed). I've decided to just rename the old async-variant to `thread_for_each()`, instead.